src/ua_securechannel.c

```diff
@@ -637,86 +638,148 @@ processSequenceNumberSym(UA_SecureChannel *channel, UA_UInt32 sequenceNumber) {
 }
 #endif
 
-/* The chunk body begins after the SecureConversationMessageHeader */
 static UA_StatusCode
-decryptAddChunk(UA_SecureChannel *channel, UA_MessageType messageType,
-                UA_ChunkType chunkType, UA_ByteString *chunk,
-                UA_Boolean allowPreviousToken) {
-    /* Has the SecureChannel timed out? */
-    if(channel->state == UA_SECURECHANNELSTATE_CLOSED)
-        return UA_STATUSCODE_BADSECURECHANNELCLOSED;
+getSecurityHeaderAlgorithms(void *application,
```

Comment:
This function is too complicated. I wouldn't know how to use it properly.
It looks as if we try to generalize a functionality that doesn't want to be generalized. ;-)

Reply:
I inlined it. Should be cleaner now.